src/App.js

```diff
@@ -1,183 +1,272 @@
+        {/*: <label htmlFor={props.id}><s className={"strikethrough"}>{props.text}</s></label> </span>*/}
+            <span onClick={() => handlePriorityClick(1, props.id, props.priority)} value={1} className={1 > props.priority ? "unchecked" : "checked"}>!   </span>
+            <span onClick={() => handlePriorityClick(2, props.id, props.priority)} value={2} className={2 > props.priority ? "unchecked" : "checked"}>!   </span>
+            <span onClick={() => handlePriorityClick(3, props.id, props.priority)} value={3} className={3 > props.priority ? "unchecked" : "checked"}>!   </span>
+            <br/><br/>
+            </span>
+    );
+}
+
+function handlePriorityClick(value, id, priority){
+    if(value === priority){
+        setDoc(doc(db, collectionName, id),
+            {"priority": value - 1}, {merge: true}).then(() => console.log("Set new priority"))
+    } else {
+        setDoc(doc(db, collectionName, id),
+            {"priority": value}, {merge: true}).then(() => console.log("Set new priority"))
     }
+}
+
+
+function handleRenaming(e, id){
+    setDoc(doc(db, collectionName, id),
+        {"text": e.target.value}, {merge: true}).then(() => console.log("Set new name"))
+    // setListItems(listItems.map(// eslint-disable-next-line
+    //     p => p.id === id ? {...p, ["text"]: e.target.value, ["value"]: e.target.value} : p))
+}
 
-    function handleAddItemClick(){
-        console.log("add item click")
+function handleToggleItemSelect(checked, selectedItems, setSelectedItems, itemId){
+    // listItems.map(item => (item.id === itemId) ? (item.checked = !item.checked) : false);
+    setDoc(doc(db, collectionName, itemId),
+        {"checked": !checked}, {merge: true}).then(() => console.log("Toggled item"))
 
-        return (<span className={"item_enter"}>
-            <label htmlFor = "item_enter"> Enter new item text:</label>
-            <br/>
-            <input type="text" id="item_enter" name="item_enter" value={curText} onChange={event => {setCurText(curText + event)}}/>
-            <button type="button" id="add_item" name="add_item" onClick={handleNewItem}>Add</button>
-        </span>);
+    if (selectedItems.includes(itemId)) {
+        setSelectedItems(selectedItems.filter(i => i !== itemId));
+    } else {
+        setSelectedItems([...selectedItems, itemId]);
     }
+}
+
+function App() {
+    // const [listItems, setListItems] = useState(data)
+
+    const [filterType, setFilterType] = useState("created_up")
+    const [reverse, setReverse] = useState(false)
+
+    const q = reverse ?
+        query(collection(db, collectionName), orderBy(filterType.slice(0, filterType.length-3), "desc"))
+        : query(collection(db, collectionName), orderBy(filterType.slice(0, filterType.length-3), "asc"));
+
+    const [listItems, loadingPage, ] = useCollectionData(q);
+
+    const [selectedItems, setSelectedItems] = useState(listItems ? listItems.filter(item => item.checked) : []);
```

Comment:
Why would you want to select the checked items to begin with?  Seems like it should always start out [].